lua/plugins/init.lua

```diff
@@ -96,13 +96,18 @@ local default_plugins = {
       vim.api.nvim_create_autocmd({ "BufRead" }, {
         group = vim.api.nvim_create_augroup("GitSignsLazyLoad", { clear = true }),
         callback = function()
-          vim.fn.system("git -C " .. '"' .. vim.fn.expand "%:p:h" .. '"' .. " rev-parse")
-          if vim.v.shell_error == 0 then
-            vim.api.nvim_del_augroup_by_name "GitSignsLazyLoad"
-            vim.schedule(function()
-              require("lazy").load { plugins = { "gitsigns.nvim" } }
-            end)
-          end
+          vim.fn.jobstart("git -C " .. '"' .. vim.fn.expand "%:p:h" .. '"' .. " rev-parse",
+            {
+              on_exit = function(_, return_code)
+                if return_code == 0 then
+                  vim.api.nvim_del_augroup_by_name "GitSignsLazyLoad"
+                  vim.schedule(function()
+                    require("lazy").load { plugins = { "gitsigns.nvim" } }
+                  end)
```

Comment:
Why schedule here again?

Reply:
Schedule runs the code on the main thread, while the job runs on an unspecified one, so to be safe, I kept the schedule call. I'm not sure it's actually necessary.

Reply:
> Why schedule here again?

it was there before too, loading gitsigns after UI would be better